cli/azd/cmd/env.go

```diff
@@ -445,10 +451,18 @@ func (ef *envRefreshAction) Run(ctx context.Context) (*actions.ActionResult, err
 	}
 	defer func() { _ = infra.Cleanup() }()
 
-	if err := ef.provisionManager.Initialize(ctx, ef.projectConfig.Path, infra.Options); err != nil {
+	// env refresh supports "BYOI" infrastructure where bicep isn't available
```

Comment:
I don't know if we should think about "external infra provider" or make `EnsureProvisionReady` a thing that providers would call internally, and possibly refine behavior such as when delete happens. Leaving all this up to #3745.